sql/src/main/java/io/crate/execution/engine/collect/stats/FilteredLogSink.java

```diff
@@ -22,36 +22,47 @@
 
 package io.crate.execution.engine.collect.stats;
 
-import io.crate.data.Input;
-import io.crate.execution.engine.collect.NestableCollectExpression;
-import io.crate.expression.InputCondition;
+import io.crate.expression.ExpressionsInput;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.Message;
+import org.elasticsearch.common.logging.Loggers;
 
 import javax.annotation.Nonnull;
 import java.util.Iterator;
-import java.util.List;
+import java.util.function.Function;
 
 public final class FilteredLogSink<T> implements LogSink<T> {
 
-    private final Input<Boolean> filter;
-    private final List<NestableCollectExpression<T, ?>> filterExpressions;
+    /**
+     * This logger name is documented in the docs for the {@link JobsLogService#STATS_JOBS_LOG_PERSIST_FILTER} setting.
+     * Take care when changing the class name or package to update the docs.
+     */
+    private static final Logger LOGGER = Loggers.getLogger(FilteredLogSink.class);
+    private final ExpressionsInput<T, Boolean> memoryFilter;
+    private final ExpressionsInput<T, Boolean> persistFilter;
+    private final Function<T, Message> createLogMessage;
     final LogSink<T> delegate;
 
-    FilteredLogSink(Input<Boolean> filter,
-                    List<NestableCollectExpression<T, ?>> filterExpressions,
+    FilteredLogSink(ExpressionsInput<T, Boolean> memoryFilter,
+                    ExpressionsInput<T, Boolean> persistFilter,
```

Comment:
I wasn't sure if it would make sense to nest 2 separate `FilteredLogSink` instances with 1 filter an "action" each. I felt like we've already enough layers and since it's related I put it together and made it concrete. But I wouldn't mind splitting it up either.

Reply:
I prefer it like that rather than splitting it up.

Reply:
I think this is currently not too complex and having both filters managed by the FilteredLogSink makes sense for now.